### PR TITLE
User locale in interactions

### PR DESCRIPTION
**Pull request by drobbins329**
Adds UserLocale and GuildLocale optional strings in the JSON converters for Interactions

### CODE REVIEW

**Comment by quinchs** [comment]
Can you add these properties to the interaction classes, ex `SocketInteraction` and `RestInteraction`

**Comment by drobbins329** [comment]
Added in SocketInteraction, RestInteraction and IDiscordInteraction. Tested both User and Guild locale string outputs in guild. Both confirmed.

File: src/Discord.Net.Core/Entities/Interactions/IDiscordInteraction.cs [hunk 1]
```diff
@@ -49,6 +49,19 @@ public interface IDiscordInteraction : ISnowflakeEntity
         /// </summary>
         IUser User { get; }
 
+        /// <summary>
+        /// The preferred locale of the invoking User.
```
Comment (quinchs) [inline src/Discord.Net.Core/Entities/Interactions/IDiscordInteraction.cs]: Add tab indenting to summaries

File: src/Discord.Net.Rest/Entities/Interactions/RestInteraction.cs [hunk 2]
```diff
@@ -32,6 +32,19 @@ public abstract class RestInteraction : RestEntity<ulong>, IDiscordInteraction
         /// </summary>
         public RestUser User { get; private set; }
 
+        /// <summary>
+        /// The preferred locale of the invoking User.
```
Comment (quinchs) [inline src/Discord.Net.Rest/Entities/Interactions/RestInteraction.cs]: Shouldn't this be `/// <inheritdoc/>`

File: src/Discord.Net.Rest/Entities/Interactions/RestInteraction.cs [hunk 3]
```diff
@@ -32,6 +32,19 @@ public abstract class RestInteraction : RestEntity<ulong>, IDiscordInteraction
...
+        /// </summary>
+        public string UserLocale { get; private set; }
+
+        /// <summary>
+        /// The preferred locale of the guild this interaction was executed in. <see cref="null"/> if not executed in a guild.
```
Comment (quinchs) [inline src/Discord.Net.Rest/Entities/Interactions/RestInteraction.cs]: Shouldn't this be `/// <inheritdoc/>`

File: src/Discord.Net.WebSocket/Entities/Interaction/SocketInteraction.cs [hunk 4]
```diff
@@ -39,6 +39,19 @@ public abstract class SocketInteraction : SocketEntity<ulong>, IDiscordInteracti
         /// </summary>
         public IDiscordInteractionData Data { get; private set; }
 
+        /// <summary>
+        /// The preferred locale of the invoking User.
```
Comment (quinchs) [inline src/Discord.Net.WebSocket/Entities/Interaction/SocketInteraction.cs]: Shouldn't this be `/// <inheritdoc/>`

File: src/Discord.Net.WebSocket/Entities/Interaction/SocketInteraction.cs [hunk 5]
```diff
@@ -39,6 +39,19 @@ public abstract class SocketInteraction : SocketEntity<ulong>, IDiscordInteracti
...
+        /// </summary>
+        public string UserLocale { get; private set; }
+
+        /// <summary>
+        /// The preferred locale of the guild this interaction was executed in. <see cref="null"/> if not executed in a guild.
```
Comment (quinchs) [inline src/Discord.Net.WebSocket/Entities/Interaction/SocketInteraction.cs]: Shouldn't this be `/// <inheritdoc/>`

**Review by quinchs — CHANGES_REQUESTED** [review]

File: src/Discord.Net.Core/Entities/Interactions/IDiscordInteraction.cs (hunk 1 above)
Comment (drobbins329) [reply src/Discord.Net.Core/Entities/Interactions/IDiscordInteraction.cs]: done

**Review by quinchs — APPROVED** [review]

**Pull request merged**